Quest.Application/Tasks/Queries/GetQuestTasksQueryHandler.cs

```diff
@@ -37,19 +37,24 @@ public GetQuestTasksQueryHandler(Db context, UserManager<ApplicationUser> userMa
             if (quest == null)
                 return BaseResponse.Failure<List<TaskAndHintsDTO>>("Could not find quest with provided id.");
 
-            if (!quest.IsReadyToReceiveTaskAttempts())
-                return BaseResponse.Failure<List<TaskAndHintsDTO>>("Quest is not in active state yet.");
-
             var user = await _context.Users.FirstOrDefaultAsync(x => x.Id == request.UserId,
                 cancellationToken: cancellationToken);
 
             if (user == null)
                 return BaseResponse.Failure<List<TaskAndHintsDTO>>("Internal: request user does not exist.");
 
+            var userIsAdmin = await _userManager.IsInRoleAsync(user, "Admin");
+            var userIsQuestCreator = request.UserId == quest.AuthorId; 
+            
+            if (!quest.IsReadyToReceiveTaskAttempts())
```

Comment:
why nested if

Reply:
it's like 55-58, and I don't like long predicates

imo, with nested if its much more readable

Reply:
you can add a line break, in 55-58 it is more like a forced decision due to separate function call

Reply:
😎